tool/parse_cli.c

```diff
@@ -0,0 +1,299 @@
+        "\t\t-v --privkey          Private key output location[default=priv.bin]\n"
+        "\t\t-b --pubkey           Public key output location[default=pub.bin]\n"
+        ;
+
+    static struct option cli_options[] =
+    {
+        {"privkey", required_argument, NULL, 'v'},
+        {"pubkey", required_argument, NULL, 'b'},
+        {"help", no_argument, NULL, 'h'},
+        {NULL, 0, NULL, 0}
+    };
+    int c;
+    while ((c = getopt_long(argc, argv, "v:b:h", cli_options, NULL)) != -1) {
+        switch (c) {
+            case 'v':
+                params->privkey=optarg;
+                break;
+            case 'b':
+                params->pubkey=optarg;
+                break;
+            case 'h':
+                fprintf(stderr, usage_str, argv[0]);
+                exit(1);
+        }
+    }
+}
+
+
+void parse_genx509cert_cli(int argc, char **argv, struct cli_params *params)
+{
+    params->cert = "cert.bin";
+    params->id = "id.bin";
+    params->privkey = "priv.bin";
+    params->pubkey = "pub.bin";
+
+    const char *usage_str = "Generate x509 certificate.\n\n"
+        "Usage: %s [-h] [-v <file>] [-b <file>] [-d <file>] [-c <file>]\n"
+        "\tOptions:\n"
+        "\t\t-h --help             Display this message.\n"
+        "\t\t-v --privkey          Private key input location[default = priv.bin]\n"
+        "\t\t-b --pubkey           Public key input location[default = pub.bin]\n"
+        "\t\t-d --id               ID input location[default=id.bin]\n"
+        "\t\t-c --certificate      Certificate output location[default=cert.bin]\n"
+        ;
+
+    static struct option cli_options[] =
+    {
+        {"privkey", required_argument, NULL, 'v'},
+        {"pubkey", required_argument, NULL, 'b'},
+        {"id", required_argument, NULL, 'd'},
+        {"certificate", required_argument, NULL, 'c'},
+        {"help", no_argument, NULL, 'h'},
+        {NULL, 0, NULL, 0}
+    };
+    int c;
+    while ((c = getopt_long(argc, argv, "v:b:d:c:h", cli_options, NULL)) != -1) {
+        switch (c) {
+            case 'v':
+                params->privkey = optarg;
+                break;
+            case 'b':
+                params->pubkey = optarg;
+                break;
+            case 'd':
+                params->id = optarg;
+                break;
+            case 'c':
+                params->cert = optarg;
+                break;
+            case 'h':
+                fprintf(stderr, usage_str, argv[0]);
+                exit(1);
+        }
+    }
+}
+
+
+void parse_wrapkeys_cli(int argc, char **argv, struct cli_params *params)
+{
+    params->privkey = "priv.bin";
+    params->pubkey = "pub.bin";
+    const char *usage_str = "Generate ASN.1 wrapped keys.\n\n"
+        "Usage: ./xtt-tool wrapkeys [-h] [-v <file>] [-b <file>] [-a <file>]\n"
+        "\tOptions:\n"
+        "\t\t-h --help             Display this message.\n"
+        "\t\t-v --priv          Private key input location[default = priv.bin]\n"
```

Comment:
Let's be consistent on spacing for the default part of the usage message.

I propose there be a space before the `[default` part, and spaces between the `=`.

What do you think?